### PR TITLE
Clean templating docs

### PR DESCRIPTION
**Pull request by sergiomsantos**
# Description

This PR refactors the `Templating` section of the docs to reflect the new implementation of the `TemplateManager` class.
It describes the basic functionalities and shows how they can be used to address the different use cases.

### CODE REVIEW

**Review by Pbarbosa92 — APPROVED** [review]

File: docs/explore_api/templating.md [hunk 1]
```diff
@@ -3,307 +3,384 @@
...
+        ├── ...
+```
 
-# Initialize the simulator and run the simulation
-splishsplash = inductiva.simulators.SplishSplash()
 
-# Initialize the templating manager
-template_manager = inductiva.TemplateManager(template_dir=input_dir,
-                                    root_dir="splishsplash-viscous")
+The `TemplateManager` class also provides a method to add external files to the
```
Comment (AugustoPeres) [inline docs/explore_api/templating.md:279]: This lines ends with nothing below it or am I missing something?

Comment (sergiomsantos) [reply docs/explore_api/templating.md]: It was supposed to be deleted. Addressed in [1b4d5b1](https://github.com/inductiva/inductiva/pull/1249/commits/1b4d5b106ec2d2381eb4242fdae6e06b662279e5).

**Review by AugustoPeres — APPROVED** [review]

File: docs/explore_api/templating.md [hunk 2]
```diff
@@ -3,307 +3,390 @@
 ## Introduction
 
 The Inductiva API is all about enabling you to simulate at scale. As we have shown,
-with a few lines of Python code, you can send your simulations to MPI Clusters assembled
-from last-generation cloud hardware, letting you run much larger simulations than
-you would be able to use your local resources. Or you can spin up a large Machine
+with a few lines of Python code, you can send your simulations to MPI Clusters
+assembled from last-generation cloud hardware, letting you run much larger simulations
+than you would be able to use your local resources. Or you can spin up a large Machine
```
Comment (IvanPombo) [inline docs/explore_api/templating.md:8]:
```suggestion
than you would be able to using your local resources. Or you can spin up a large Machine
```

File: docs/explore_api/templating.md [hunk 3]
```diff
@@ -3,307 +3,390 @@
...
-            "translation": [0, 0, 0],
-            "scale": [0.5, 0.5, 0.5],
-            "initialVelocity": [0, 0, 0]
-        }
-    ]
-}
+what you need when you want to find the optimal solution for a problem, and the
+only way to test each candidate's solution is by simulating it. To do so, one needs
+to configure multiple simulations, each with a slightly different set of parameters.
+Generally speaking, the goal is to explore the largest possible volume of the
```
Comment (IvanPombo) [inline docs/explore_api/templating.md:14]: Volume here seems weird, and in fact, the goal is to explore the design space. The solution refers to results of the simulation you are obtaining.

File: docs/explore_api/templating.md [hunk 4]
```diff
@@ -3,307 +3,390 @@
...
+solution space.
+
+In this context, the Inductiva API provides a powerful tool for exploring the solution
```
Comment (IvanPombo) [inline docs/explore_api/templating.md:17]:
```suggestion
In this context, the Inductiva API provides a powerful tool for exploring the design
```

**Review by IvanPombo — APPROVED** [review]
Just left a few comments at the start! Else, the code looks fine! LGTM!

**Pull request merged**